Sources/NIOHTTP2/ContentLengthVerifier.swift

```diff
@@ -48,8 +48,19 @@ extension ContentLengthVerifier {
 }
 
 extension ContentLengthVerifier {
-    internal init(_ headers: HPACKHeaders) {
-        self.expectedContentLength = headers[canonicalForm: "content-length"].first.flatMap { Int($0, radix: 10) }
+    internal init(_ headers: HPACKHeaders) throws {
+        var headerIterator = headers[canonicalForm: "content-length"].makeIterator()
+        guard let first = headerIterator.next() else {
+            return
+        }
+
+        // multiple content-length headers are permitted as long as they agree
+        for header in headerIterator {
+            guard header == first else {
+                throw NIOHTTP2Errors.contentLengthHeadersMismatch()
+            }
+        }
```

Comment:
TIL that `for ... in` works with an iterator as well 🤯
However, personally I like to use the collection algorithms instead but thats just a style preference so feel free to ignore:
```suggestion
        let contentLengths = headers[canonicalForm: "content-length"]
        guard let first = a.first else {
            return
        }
        // multiple content-length headers are permitted as long as they agree
        guard contentLengths.dropFirst().allSatisfy({ $0 == first }) else {
            throw NIOHTTP2Errors.contentLengthHeadersMismatch()
        }
```

Reply:
I like your version, it's more succinct 🙏